controllers/testclustertopology_controller.go

```diff
@@ -626,11 +626,11 @@ func (r *TestClusterTopologyReconciler) updateTiDBCluster(ctx context.Context, c
 			return false, ctrl.Result{RequeueAfter: 10 * time.Second}, nil
 		}
 	}
-	if cluster.IsServerConfigModified(ct.Status.PreTiDBCluster.ServerConfigs, ct.Spec.TiDBCluster.ServerConfigs) {
+	if cluster.IsServerConfigModified(ct.Status.PreTiDBCluster.ServerConfigs, ct.Spec.TiDBCluster.ServerConfigs) || cluster.IsComponentsConfigModified(ct.Status.PreTiDBCluster, ct.Spec.TiDBCluster) {
```

Comment:
What's the difference between `cluster.IsComponentsConfigModified` and `checkImmutableFieldChanged`? IMO you could retain only one of them.

Reply:
done